frontend/javascripts/oxalis/model/sagas/isosurface_saga.js

```diff
@@ -296,14 +321,50 @@ function* removeIsosurface(action: RemoveIsosurfaceAction): Saga<void> {
   }
 }
 
+function* noteEveryEditedCell(): Saga<void> {
+  const activeCellId = yield* select(state => enforceVolumeTracing(state.tracing).activeCellId);
+  modifiedCells.add(activeCellId);
+}
+
+function* refreshIsosurfaces(): Saga<void> {
+  const renderIsosurfaces = yield* select(state => state.datasetConfiguration.renderIsosurfaces);
+  if (!renderIsosurfaces) {
+    return;
+  }
+  yield* put(saveNowAction());
+  // We reload all cells that got modified till the start of reloading.
+  // By that we avoid that removing cells that got annotated during reloading from the modifiedCells set.
+  const currentlyModifiedCells = new Set(modifiedCells);
+  modifiedCells.clear();
+  // First create an array containing information about all loaded isosurfaces as the map is manipulated within the loop.
+  for (const [cellId, threeDMap] of Array.from(isosurfacesMap.entries())) {
+    if (!currentlyModifiedCells.has(cellId)) {
+      continue;
+    }
+    const isoSurfacePositions = threeDMap.entries().filter(([value, _position]) => value);
+    if (isoSurfacePositions.length <= 0) {
+      continue;
+    }
+    // Removing Isosurface from cache.
+    yield* call(removeIsosurface, removeIsosurfaceAction(cellId), false);
+    // The isosurface should only be removed once after re-fetching the isosurface first position.
+    let shouldBeRemoved = true;
+    for (const [, position] of isoSurfacePositions) {
+      // Reload the Isosurface.
+      yield* call(ensureSuitableIsosurface, null, position, cellId, shouldBeRemoved);
+      shouldBeRemoved = false;
+    }
+  }
+  yield* put(finishedRefreshingIsosurfacesAction());
+}
+
 export default function* isosurfaceSaga(): Saga<void> {
   yield* take("WK_READY");
   yield _takeEvery(FlycamActions, ensureSuitableIsosurface);
-  yield _takeEvery(
-    ["CHANGE_ACTIVE_ISOSURFACE_CELL", "SET_ACTIVE_CELL"],
-    changeActiveIsosurfaceCell,
-  );
+  yield _takeEvery("CHANGE_ACTIVE_ISOSURFACE_CELL", changeActiveIsosurfaceCell);
   yield _takeEvery("TRIGGER_ISOSURFACE_DOWNLOAD", downloadActiveIsosurfaceCell);
   yield _takeEvery("IMPORT_ISOSURFACE_FROM_STL", importIsosurfaceFromStl);
   yield _takeEvery("REMOVE_ISOSURFACE", removeIsosurface);
+  yield _takeEvery("REFRESH_ISOSURFACES", refreshIsosurfaces);
```

Comment:
When the refresh button is clicked, I think it would also make sense to fetch the isosurfaces at the current position (as a seed) with the cell id at that position. That way, the button can also be used to render the very first isosurface without needing to know about shift+click.